server/controllers/profileController.js

```diff
@@ -0,0 +1,110 @@
+const Profile = require("../models/ProfileModel");
+const User = require("../models/User");
+const mongoose = require("mongoose");
+
+const toObjectId = mongoose.Types.ObjectId;
+
+// create new profile using given arguments
+
+exports.createProfile = async (req, res, next) => {
+  try {
+    const { firstName, lasttName, phone, address, gender, description } =
+      req.body;
+
+    const userId = req.user.id;
+    const user = await User.findById(userId);
+    if (!user) {
+      res.status(401).json({ message: "Not Authorized" });
+    }
+
+    if (
+      firstName === undefined ||
```

Comment:
In the future, you can create a function to validate data so that it can be reused for different routes. Also, this can further be simplified to `if (!firstName || !lastName....)` 😄